pingpong/src/main/java/com/cedrus/pingpong/kafka/PingPongStream.java

```diff
@@ -20,39 +20,74 @@
 import java.io.IOException;
 import java.util.Properties;
 import java.util.Random;
+import java.util.UUID;
 
 
 @Slf4j
 @Component
 public class PingPongStream {
-    @Autowired
     private KafkaConfig kafkaConfig;
-    @Autowired
     private TopicConfig topicConfig;
-    @Autowired
     private AppConfig appConfig;
+    private ObjectMapper objectMapper;
 
-    public void startListening(String topic) throws IOException {
-        log.info("==== STARTED LISTENING ON " + topic + " ====");
-        createStream(topic);
+    @Autowired PingPongStream(KafkaConfig kafkaConfig,
+                              TopicConfig topicConfig,
+                              AppConfig appConfig,
+                              ObjectMapper objectMapper)
+    {
+        this.kafkaConfig = kafkaConfig;
+        this.topicConfig = topicConfig;
+        this.appConfig = appConfig;
+        this.objectMapper = objectMapper;
     }
 
-    public void createStream(String topic) {
+    public void startStream(String teamGroupId, String playerId) throws IOException {
+
+        log.info("==== STARTED STREAM ON " + topicConfig.getPingPong() + " ====");
+        Serde<String> stringSerde = Serdes.String();
+
+        buildStream(teamGroupId, stringSerde, playerId);
+    }
+
+    private void transformAndProduceStream(KStream stream, Serde serde, String playerId) {
+        KStream<String, String> mappedValuesStream = stream.transformValues(createMessageProcessor(playerId));
+        mappedValuesStream.to(topicConfig.getPingPong(), Produced.with(serde, serde));
+    }
+
+    private void buildStream(String teamGroupId, Serde serde, String playerId) {
         Properties props = new Properties();
         props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaConfig.getBootstrapServers());
-        props.put(StreamsConfig.APPLICATION_ID_CONFIG, kafkaConfig.getKafkaAppId() + topic);
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, teamGroupId);
 
         StreamsBuilder builder = new StreamsBuilder();
-        Serde<String> stringSerde = Serdes.String();
-        KStream stream = builder.stream(topic, Consumed.with(stringSerde, stringSerde));
-        String newTopic = topic.equals(topicConfig.getPing()) ? topicConfig.getPong() : topicConfig.getPing();
-        KStream<String, String> mappedValues = stream.transformValues(processMessage());
-        mappedValues.to(newTopic, Produced.with(stringSerde, stringSerde));
+
+        KStream stream = builder.stream(topicConfig.getPingPong(), Consumed.with(serde, serde));
+        stream.selectKey(new KeyValueMapper() {
+            @Override
+            public Object apply(Object key, Object value) {
+                UUID random = UUID.randomUUID();
+                return random;
+            }
+        });
+        KStream[] branches = stream.branch((key, value) -> {
+            PingPongMessage message = new PingPongMessage();
```

Comment:
Why initialize if you're just going to dereference? Maybe make this null instead.